integration_test/ginkgo_integration_test.go

```diff
@@ -0,0 +1,61 @@
+package integrationtest
+
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openebs/node-disk-manager/integration_test/minikube_adm"
+	"github.com/openebs/node-disk-manager/integration_test/ndm_util"
+)
+
+func TestIntegrationNDM(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Integration Test Suit")
+}
+
+var _ = BeforeSuite(func() {
+	// It starts minikube if it is not Running
+	minikubeadm.Setup()
+
+	// It waits till namespace is ready
+	ndmutil.WaitTillDefaultNSisReady()
+
+	// It applies our CRDs
+	ndmutil.ApplyCRDs()
+
+	// It prepares configuration and Applies the same
+	ndmutil.PrepareAndApplyYAML()
+
+	// It waits till node-disk-manager is ready
+	ndmutil.WaitTillNDMisUp()
+})
+
+var _ = AfterSuite(func() {
+	// It Delete minikube if it is running
+	// It removes residue containers
+	// It removes remaining residue files
+	ndmutil.Clean()
+})
+
+var _ = Describe("Integration Test", func() {
+	// Now as BeforeSuit has run, We shall have a healthy node-disk-manager daemonset
+	When("We check the log", func() {
+		It("has `Started the controller` in the log", func() {
```

Comment:
can we rephrase?
It("should contain `started the controller` in the log")

Reply:
I am really not a big fan of checking logs to verify things esp. when we are doing automation.
There is no hard & fast rules though.
However, we must make use of better verification techniques.

Reply:
Currently node-disk-manager produces very few logs, so I have created an initial function to validate the log to mimic the way we search for a problem this might be one (of the many) tool.

Reply:
I request you to see line 116 of the same file, I have mentioned there that `ValidateNdmLog` function to validate the log is tend to grow as our code grows.